ftl/php/builder.py

```diff
@@ -19,62 +19,92 @@
 import logging
 import datetime
 
-from containerregistry.client.v2_2 import append
 from containerregistry.transform.v2_2 import metadata
+from containerregistry.client.v2_2 import docker_digest
 
 from ftl.common import builder
 from ftl.common import ftl_util
+from ftl.common import build_layer
 
-_PHP_NAMESPACE = 'php-composer-lock-cache'
+_PHP_NAMESPACE = 'php-package-lock-cache'
 _COMPOSER_LOCK = 'composer.lock'
 _COMPOSER_JSON = 'composer.json'
 
 
-class PHP(builder.JustApp):
-    def __init__(self, ctx):
-        self.descriptor_files = [_COMPOSER_LOCK, _COMPOSER_JSON]
-        self.namespace = _PHP_NAMESPACE
-        super(PHP, self).__init__(ctx)
+class PHP(builder.RuntimeBase):
+    def __init__(self, ctx, args, cache_version_str):
+        super(PHP, self).__init__(ctx, _PHP_NAMESPACE, args, cache_version_str,
+                                  [_COMPOSER_LOCK, _COMPOSER_JSON])
+
+    def Build(self):
+        lyrs = []
+        lyrs.append(self._base)
+        if ftl_util.has_pkg_descriptor(self._descriptor_files,
+                                       self._ctx):
+            pkg = self.PackageLayer(self._ctx,
+                                    self._descriptor_files, None,
+                                    self._args.destination_path)
+            if not pkg.IsCached(self._cash, self._base,
+                                        self._namespace):
+                pkg.BuildLayer()
+                pkg.CacheLayerImage(self._cash, self._base, self._namespace)
+            lyrs.append(pkg)
+
+        app = self.AppLayer(self._ctx)
+        app.BuildLayer()
+        lyrs.append(app)
+        ftl_image = self.BuildImageFromLayerImages(lyrs)
```

Comment:
BuildImageFromLayerImages kind of stutters. Maybe "AppendLayersIntoImage" or something?

Reply:
done